core/lib/types/src/transaction_request.rs

```diff
@@ -826,8 +826,10 @@ impl L2Tx {
         let meta = value.eip712_meta.take().unwrap_or_default();
         validate_factory_deps(&meta.factory_deps)?;
 
-        if value.to.is_none() && !allow_no_target {
-            return Err(SerializationTransactionError::ToAddressIsNull);
+        if value.to.is_none() {
+            if !allow_no_target || value.is_eip712_tx() {
```

Comment:
Nit: Please used a collapsed condition `value.to.is_none() && (!allow_no_target || value.is_eip712_tx())`; clippy thinks it's serious business™️.